### PR TITLE
Publish to Maven Central

### LINKED ISSUE #45: Move to Maven Central
jcenter will be closed soon and this artifact won't be available anymore. Please, move it to the Maven Central

### PR DESCRIPTION
**Pull request by jeggy**
closes #45

Tasks:
* [x] Setup SonaType account and get `com.apurebase` approved there
* [x] Gradle configuration for publishing to Maven Central
* [x] Make sure it's using 1.8
* [x] Setup Jupiter again
* [x] Create GitHub action to publish to Maven Central
* [x] Publish old versions to Maven Central

### CODE REVIEW

File: arkenv-yaml/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,91 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    base
+    kotlin("jvm") version "1.3.72"
```
Comment (AndreasVolkmann) [inline arkenv-yaml/build.gradle.kts]: Any reason for lower version?

**Comment by AndreasVolkmann** [comment]
Nice to see `.kts` build files 👍

**Comment by AndreasVolkmann** [comment]
The errors come from `Cannot inline bytecode built with JVM target 1.8 into bytecode that is being built with JVM target 1.6.`

We need something like this: 
```kotlin
compileOptions {
    sourceCompatibility = 1.8
    targetCompatibility = 1.8
}
tasks.withType(org.jetbrains.kotlin.gradle.tasks.KotlinCompile).all {
    kotlinOptions {
        jvmTarget = "1.8"
    }
}
```

File: arkenv-yaml/build.gradle.kts (hunk 1 above)
Comment (jeggy) [reply arkenv-yaml/build.gradle.kts]: I just upgraded to the latest 1.3.x version. We can upgrade to 1.4 in some other branch 😄

**Comment by jeggy** [comment]
This is still in `WIP` 😄 

But yeah, current things I want to fix is:
* Make sure it's using 1.8
* Setup Jupiter again
* Create GitHub action to publish to SonaType

File: arkenv-yaml/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+val isReleaseVersion = !version.toString().endsWith("SNAPSHOT")
+
+//val sourceTests: SourceSetOutput =
+
+dependencies {
+    implementation(kotlin("stdlib-jdk8"))
+    compile(project(":arkenv"))
+    compile("org.yaml:snakeyaml:$snakeyamlVersion")
+
+    // Can't figure out how to extend via arkenv project test sourcesets
```
Comment (jeggy) [inline arkenv-yaml/build.gradle.kts]: TODO: Forgot to remove this comment :smile:

**Comment by jeggy** [comment]
There I finally got Kotlin Gradle Script to fully play along :laughing: 

As some test classes were shared between `:arkenv` and `:arkenv-yaml` I have converted this into a testFixture project and moved all shared utilities into `arkenv/src/testFixtures` directory :+1: 

After this has been merged, releases will be automatically published to MavenCentral when a github release has been made, which can be see in `.github/workflows/maven-publish.yml`

**Review by AndreasVolkmann — APPROVED** [review]

**Pull request merged**